### PR TITLE
Congrats: remove legacy thank you code

### PR DESCRIPTION
**Pull request by chad1008**
Related to 6015-gh-Automattic/dotcom-forge

## Proposed Changes

Remove legacy unused/unreachable thank you code from [client/my-sites/checkout/checkout-thank-you/index.tsx](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/index.tsx)

There is additional code to remove in other files that will be split up/batched into followup(s).

## Testing Instructions

Confirm that these purchase flows continue to match behavior on `trunk`:

- Domain only
- Dlan only
- Ecomm plan only
- Plan with domain 
- Add-ons (e.g. Jetpack AI)
- Delayed domain transfer (i.e. select a domain transfer during signup)
- Titan email
- Google Workspace email
- Bulk domain transfer (e.g. https://wordpress.com/transfer-google-domains/)

All but Delayed domain transfers should render the new Congrats pages.

### CODE REVIEW

**Comment by stephanethomas** [comment]
It appears that both unit tests and E2E tests have failed. More importantly, this pull request removes a significant amount of code, some of which including conditional branches, which makes it difficult for me to be confident that we're ditching the right things. I'm reluctant to test these changes against each mentioned flow, as it will be not only time-consuming but also risky, given that we might miss a specific flow.

I'd like to suggest a different approach where we record a Tracks event at the beginning of each major block of code we plan to remove, and let that run for a week or two. This will allow us to confirm that these code paths are indeed no longer in use. What do you think? Or perhaps I'm being too risk-averse 🙂?

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/index.tsx:687]: Noting that, [`isRefactoredForThankYouV2()`](https://github.com/Automattic/wp-calypso/blob/456cc1137f8a1d90ccd2778eed4d14ada2986ff2/client/my-sites/checkout/checkout-thank-you/redesign-v2/utils.ts#L11) returns `true` for all scenarios except for a failed order or delayed transfer are considered redesigned. 

So domain-only purchases will enter the [refactored logic](https://github.com/Automattic/wp-calypso/blob/456cc1137f8a1d90ccd2778eed4d14ada2986ff2/client/my-sites/checkout/checkout-thank-you/index.tsx#L590), and the page will be rendered by: https://github.com/Automattic/wp-calypso/blob/05caf22fe6e33b861c937eaabf113616d9fe7896/client/my-sites/checkout/checkout-thank-you/index.tsx#L604-L605

The block of condition starting at L687 is never executed.

**Comment by wongasy** [comment]
>both unit tests and E2E tests have failed. 

Those tests have been failing since yesterday (see p1712772550012179-slack-C02DQP0FP), but it should be fixed by https://github.com/Automattic/wp-calypso/pull/89420. A rebase should address the issue in this PR!

As for the unit tests, @chad1008, you might need to remove tests that are no longer relevant.

> More importantly, this pull request removes a significant amount of code, some of which including conditional branches, which makes it difficult for me to be confident that we're ditching the right things. I'm reluctant to test these changes against each mentioned flow, as it will be not only time-consuming but also risky, given that we might miss a specific flow.

@stephanethomas I have full context on these changes and am fairly confident about removing the code. I'll take over the review :) I've also added inline comments to explain why we're removing the code.

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/index.tsx]: @chad1008 We might want to keep this for handling failed purchases.

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/index.tsx]: Since we're keeping the legacy standard thank you page (see previous comment), this function should be kept.

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/index.tsx]: Noting that this is the placeholder that @nightnei will be updating to align with the redesigned congrats page as part of this issue https://github.com/Automattic/dotcom-forge/issues/6500

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/index.tsx]: We should also keep this function, but remove all conditions except for failed purchases. 

Congrats page for plans is handled by the `<PlanOnlyThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/05caf22fe6e33b861c937eaabf113616d9fe7896/client/my-sites/checkout/checkout-thank-you/index.tsx#L607-L613

Congrats page for Google Workspace is handled by the `<GoogleWorkspaceSetupThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/456cc1137f8a1d90ccd2778eed4d14ada2986ff2/client/my-sites/checkout/checkout-thank-you/index.tsx#L636-L639

Congrats page for Titan is handled by the `<TitanSetupThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/456cc1137f8a1d90ccd2778eed4d14ada2986ff2/client/my-sites/checkout/checkout-thank-you/index.tsx#L630-L635

Congrats page for all other purchases will be handled by the `<GenericThankYou>` component:
https://github.com/Automattic/wp-calypso/blob/05caf22fe6e33b861c937eaabf113616d9fe7896/client/my-sites/checkout/checkout-thank-you/index.tsx#L641

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (wongasy) [inline client/my-sites/checkout/checkout-thank-you/index.tsx]: Similarly, this function should be kept, but with all conditions except for `hasFailedPurchases` removed.

**Review by wongasy — CHANGES_REQUESTED** [review]

**Comment by chad1008** [comment]
Thanks for the reviews (both of them!)

@wongasy I believe I've addressed your feedback regarding the functions we'll want to keep, but would love a second pair of eyes. I'm going to let CI run and then look into tracking down any obsolete unit tests.

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (nightnei) [reply client/my-sites/checkout/checkout-thank-you/index.tsx]: If we go with my approach - we can remove it. Details in my PR with the proposed changes https://github.com/Automattic/wp-calypso/pull/89487#discussion_r1562697973

**Review by wongasy — COMMENTED** [review]
We're almost there! Once the following issues are addressed, we should be good to go!

* The white background is gone for the Generic Congrats Page. That can be addressed with this fix: https://github.com/Automattic/wp-calypso/pull/88359/files#diff-ca8bf6f94e9100bf77a11d24ea3c90dc5ccffd14427a6efac01280f20b4c0e2aR666

Screen Shot 2024-04-15 at 1 50 30 PM

* There's a bug in production where the "Need extra help" link is hugging the edge of the page. Let's take the opportunity to fix that!

Screen Shot 2024-04-15 at 2 06 38 PM

* There's another a bug in production where we have extra padding on mobile. Let's also take the opportunity to fix that!

Screen Shot 2024-04-15 at 2 12 54 PM

The last two issues should be addressed with the following diff:

```diff
--- a/client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/index.tsx
+++ b/client/my-sites/checkout/checkout-thank-you/redesign-v2/masterbar-styled/index.tsx
@@ -8,6 +8,7 @@ import { DefaultMasterbarContact } from './default-contact';
 const MasterbarStyledBlock = styled( Masterbar )`
        --color-masterbar-background: var( --studio-white );
        --color-masterbar-text: var( --studio-gray-60 );
+       padding-right: 24px;
        border-bottom: 0;
 `;
 
diff --git a/client/my-sites/checkout/checkout-thank-you/style.scss b/client/my-sites/checkout/checkout-thank-you/style.scss
index e504af752d..fd09910f8b 100644
--- a/client/my-sites/checkout/checkout-thank-you/style.scss
+++ b/client/my-sites/checkout/checkout-thank-you/style.scss
@@ -28,7 +28,7 @@
 
                &.main {
                        max-width: 750px;
-                       padding: 0 24px 0;
+                       padding: 0;
                        display: flex;
                        flex-direction: column;
                        text-align: initial;
@@ -663,6 +663,15 @@
        }
 }
```

File: client/my-sites/checkout/checkout-thank-you/index.tsx
Comment (chad1008) [reply client/my-sites/checkout/checkout-thank-you/index.tsx]: Excellent, thanks for cross linking... I'll watch your PR and plan on rebasing/updating as needed when yours merges @nightnei

Comment (wongasy) [reply client/my-sites/checkout/checkout-thank-you/index.tsx]: @chad1008 Initially, I thought the placeholder PR will be merged first, but that didn't turn out to be the case. If you're able to address the final feedback and get this PR shipped today, then Vova can rebase his PR :)

Comment (chad1008) [reply client/my-sites/checkout/checkout-thank-you/index.tsx]: Just finished now and was about to confirm whether or not Vova's PR also removed this placeholder 😄

**Comment by chad1008** [comment]
I've applied the CSS fixes and removed that legacy placeholder that @nightnei's PR will be replacing, so when that one is rebased it won't be there cluttering things up any longer.

One more review and I think we'll be in good shape! cc @wongasy

**Comment by wongasy** [comment]
>I've applied the CSS fixes and removed that legacy placeholder that @nightnei's PR will be replacing, so when that one is rebased it won't be there cluttering things up any longer.

@chad1008 We should not remove that placeholder, if the idea is to ship this PR before Vova's 😅 Otherwise, we'll be left with no placeholder.

**Comment by chad1008** [comment]
> We should not remove that placeholder, if the idea is to ship this PR before Vova's 😅 Otherwise, we'll be left with no placeholder.

🤦 yeah, let's pretend I didn't say any of that please. I've undone my foolish fix. CI running now.

**Comment by chad1008** [comment]
current e2e failures look unrelated, and have been failing consistently on other PRs for the last half hour or so. I'll check back periodically and re-run until we get a green light

**Review by wongasy — APPROVED** [review]
LGTM and tests well!

I've tested the Congrats page for the following purchases:

* Domain only

Screen Shot 2024-04-15 at 8 40 22 PM

* Domain + Email + Plan

Screen Shot 2024-04-15 at 7 42 42 PM

* Domain + Titan

Screen Shot 2024-04-15 at 7 53 07 PM

* Multiple Titan mailboxes

Screen Shot 2024-04-15 at 7 43 20 PM

* Google Workspace

Screen Shot 2024-04-15 at 7 43 06 PM

* Google Workspace additional mailbox

Screen Shot 2024-04-15 at 7 42 55 PM

* E-Commerce plan 

Screen Shot 2024-04-15 at 7 44 57 PM

* Domain transfer

Screen Shot 2024-04-15 at 7 46 33 PM

**Comment by wongasy** [comment]
>current e2e failures look unrelated

Hm... I've flagged it in the #calypso channel p1713226175237849-slack-C02DQP0FP

**Pull request merged**
